File: Snooper/src/androidTest/java/com/prateekj/snooper/rules/DataResetRule.java
package com.prateekj.snooper.rules;


import android.database.sqlite.SQLiteDatabase;

import com.prateekj.snooper.database.SnooperDbHelper;

import org.junit.rules.TestRule;
import org.junit.runner.Description;
import org.junit.runners.model.Statement;

import java.util.Arrays;
import java.util.List;

import static androidx.test.InstrumentationRegistry.getTargetContext;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.HEADER_TABLE_NAME;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.HEADER_VALUE_TABLE_NAME;
import static com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.HTTP_CALL_RECORD_TABLE_NAME;

public class DataResetRule implements TestRule {

  private final SnooperDbHelper snooperDbHelper;

  public DataResetRule() {
    snooperDbHelper = SnooperDbHelper.getInstance(getTargetContext());
  }

  @Override
  public Statement apply(final Statement base, Description description) {
    return new Statement() {
      @Override
      public void evaluate() throws Throwable {
        try {
          base.evaluate();
        } finally {
          SQLiteDatabase database = snooperDbHelper.getWritableDatabase();
          List<String> tableToDelete = Arrays.asList(HEADER_VALUE_TABLE_NAME, HEADER_TABLE_NAME, HTTP_CALL_RECORD_TABLE_NAME);
          for (String table : tableToDelete) {
            database.delete(table, null, null);
          }
          database.close();
        }
      }
    };
  }
}

File: Snooper/src/androidTest/java/com/prateekj/snooper/rules/DataResetRule.kt
package com.prateekj.snooper.rules


import androidx.test.InstrumentationRegistry.getTargetContext
import com.prateekj.snooper.database.SnooperDbHelper
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.HEADER_TABLE_NAME
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.HEADER_VALUE_TABLE_NAME
import com.prateekj.snooper.networksnooper.database.HttpCallRecordContract.Companion.HTTP_CALL_RECORD_TABLE_NAME
import org.junit.rules.TestRule
import org.junit.runner.Description
import org.junit.runners.model.Statement

class DataResetRule : TestRule {

  private val snooperDbHelper: SnooperDbHelper = SnooperDbHelper.getInstance(getTargetContext())

  override fun apply(base: Statement, description: Description): Statement {
    return object : Statement() {
      @Throws(Throwable::class)
      override fun evaluate() {
        try {
          base.evaluate()
        } finally {
          val database = snooperDbHelper.writableDatabase
          val tableToDelete = listOf(
            HEADER_VALUE_TABLE_NAME,
            HEADER_TABLE_NAME,
            HTTP_CALL_RECORD_TABLE_NAME
          )
          for (table in tableToDelete) {
            database.delete(table, null, null)
          }
          database.close()
        }
      }
    }
  }
}
